chapters/annotations.tex

```diff
@@ -2324,6 +2324,16 @@ \subsection{Licensing}\label{licensing}
 \end{example}
 
 
+\section{License Texts}\label{license-texts}
+
+For a top-level class the {\lstinline!annotation(License="modelica:/ModelicaLibraryName/Resources/Licenses/MyLicense.txt")!}\annotationindex{License}, gives a license file for the class.
+The annotation may give a scalar value or an array of values.
+The license files are human-readable files containing the license conditions.
+If the annotation specifies an array of license files the conditions in all of them must be satisfied (i.e., it is not intended for dual-licensing).
+External functions may also contain this annotation (with similar content), see \cref{annotations-for-external-libraries-and-include-files}.
+When generating a distributable version for a translated model the license files for used functions and top-level classes should be included.
```

Comment:
Is it a conscious decision to _not_ require that tools include the license files?

Reply:
There are several possible reasons:
1. The vagueness of "distributable version" makes it odd to have a hard requirement. (Is an executable distributable if it can be copied or does it require more? Is it distributed if distributed internally in a company?)
2. A tool may have options to handle the license files in other ways; if you generate a distributable version for embedding in a device the license files should be delivered in a completely different way: verified by a legal department, and some of the license files might be skipped (after review).
3. If the library is internal to the ones translating the model it makes sense that they license owners can skip their own licenses.

In all cases one could just remove the license files afterwards, so people can anyway circumvent it.

Reply:
Note that 'should' doesn't mean that it can be skipped at will, just that there _may_ exist valid reasons in particular circumstances to ignore a particular item, but the full implications must be understood and carefully weighed before choosing a different course.

Reply:
A nice way to resolve this would be to include a non-normative explanation of why a tool might choose to not include the license files.

Reply:
-> @HansOlsson write explanation.

Reply:
@HansOlsson ping

Reply:
Have done

Reply:
OK, let's continue at the added content.  Resolving.